core_impl/src/main/java/com/google/instrumentation/stats/StatsManagerImpl.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import java.util.Set;
+
+/**
+ * Native Implementation of {@link StatsManager}.
+ */
+public class StatsManagerImpl extends StatsManager {
+
+  @Override
+  public void registerView(ViewDescriptor viewDescriptor) {
+    Set<View> viewSet = MeasurementDescriptorToViewMap.getMap()
+        .get(viewDescriptor.getMeasurementDescriptor());
+    if (viewSet == null) {
```

Comment:
Checking that will need to iterate all the views in the set in a loop, and I think this is for registering new views so the ViewDescriptor should not match any View in the set. I'll add the checking in else{} block.